CodeEdit/Documents/WorkspaceDocument.swift

```diff
@@ -268,12 +268,12 @@ import CryptoKit
     }
 
     private func readSelectionState() throws -> WorkspaceSelectionState {
-        guard let path = fileURL?.path,
-              let pathData = path.data(using: .utf8) else { return selectionState }
-        let hash = String(CryptoKit.SHA256.hash(data: pathData).hashValue)
-        if let data = UserDefaults.standard.value(forKey: hash) as? Data {
-            let state = try PropertyListDecoder().decode(WorkspaceSelectionState.self, from: data)
-            return state
+        guard let path = fileURL?.path else { return selectionState }
+        if let hash = path.sha256Hash {
+            if let data = UserDefaults.standard.value(forKey: hash) as? Data {
+                let state = try PropertyListDecoder().decode(WorkspaceSelectionState.self, from: data)
+                return state
+            }
         }
         return selectionState
```

Comment:
combine `if let` statements with `guard let` statement for cleaner code (see other comment).